### DISCUSSION TITLE
Set menu items to 'inactive'

### DISCUSSION THREAD

**Discussion by kla-ko** [GeorgianStan/vanilla-context-menu#37] [Ideas & Suggestions]
Hi
It's a common practice for context menus to show those menu items as inactive (= grey color, not selectable), if it does not make sense to execute the action related to these menu items.
Examples: A ``paste`` action shall be inactive if there is nothing to paste in the clipboard. A ``back`` action shall be inactive if this is the first step of a process.

I don't think this is achievable today. Might be a feature for the ``updateOptions``-API ?

Regards

**Comment by GeorgianStan** [comment]
I understand. I think this can be achieved if the API of the library is updated to expose a hook that will be executed right before the context-menu is displayed and in this hook can be defined the logic that decides what items `active` or `inactive` are set via `updateOptions` method.
